### PR TITLE
chore: use Python 3.8 for docs / docfx sessions

### LINKED ISSUE #124: bug: docs build fails because there isn't Python 3.7 interpreter found
From the recent release docs build log:
```
nox > Running session docs
nox > Session docs skipped: Python interpreter 3.7 not found.
```
Noxfile for this repo doesn't seem to be templated, updating the noxfile should resolve this issue. Happy to manually release the docs once noxfile is updated and everything works fine unless docs are not required for this client library.

### PR DESCRIPTION
**Pull request by tseaver**
Works around recent changes to the Kokoro 'docs-presubmt' configuration.

See: https://github.com/googleapis/python-api-core/pull/224

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -106,7 +106,7 @@ def cover(session):
     session.run("coverage", "erase")
 
 
-@nox.session(python=DEFAULT_PYTHON_VERSION)
+@nox.session(python="3.8")
 def docs(session):
```
Comment (dandhlee) [inline noxfile.py:110]: Could you bump the sphinx version to 4.0.1?

File: noxfile.py [hunk 2]
```diff
@@ -129,7 +129,7 @@ def docs(session):
     )
 
 
-@nox.session(python=DEFAULT_PYTHON_VERSION)
+@nox.session(python="3.8")
 def docfx(session):
```
Comment (dandhlee) [inline noxfile.py:133]: Could you also pin sphinx version to 4.0.1 and use `gcp-sphinx-docfx-yaml` instead of `sphinx-docfx-yaml`?

**Review by dandhlee — CHANGES_REQUESTED** [review]
minor noxfile changes, please see below.

File: noxfile.py (hunk 1 above)
Comment (tseaver) [reply noxfile.py:110]: 2b549a1

File: noxfile.py (hunk 2 above)
Comment (tseaver) [reply noxfile.py:133]: 2b549a1

**Comment by tseaver** [comment]
@dandhlee PTAL.

**Review by dandhlee — APPROVED** [review]
LGTM, thank you!

**Pull request merged**